setup-macos.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# Fail if command failed, variable is unassigned, also show executed commands
+set -eux
+
+echo "Install Boehm GC, RE2"
+brew install bdw-gc re2
+
+echo "Install argonaut from source"
+rm -rf argonaut
+git clone https://github.com/argonaut-io/argonaut.git
```

Comment:
Can we hardcode a commit here so that builds are reproducible? i.e. `git checkout <rev>`

Reply:
Good idea - this is now done against latest Argonaut `master` at the time of writing: https://github.com/argonaut-io/argonaut/commit/2c719f155744881d30fc932dcbbf597a9ce8084c